ocs_ci/ocs/benchmark_operator.py

```diff
@@ -0,0 +1,213 @@
+        self.pgsql_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind="namespace")
+        self.pod_obj = OCP(namespace=BMO_NAME, kind="pod")
+        # list of worker nodes to label
+        self.worker_nodes = get_worker_nodes()
+        self._clone_operator()
+        self.dir += f"/{BMO_NAME}"
+
+        # to use the cache dropping pod, worker nodes need to be labeled.
+        log.info("Labeling the worker nodes for cache-dropping enable.")
+        try:
+            helpers.label_worker_node(
+                self.worker_nodes, label_key=BMO_LABEL, label_value="yes"
+            )
+        except CommandFailed:
+            # this is probably because of the nodes are already labeled, so,
+            # checking if nodes are labeled and continue anyway.
+            result = self.pod_obj.exec_oc_cmd(f"get node -l {BMO_LABEL}")
+            found = [node for node in self.worker_nodes if re.search(node, result)]
+            if len(found) == len(self.worker_nodes):
+                log.info("All worker nodes are labeled")
+            else:
+                log.warning("Labeling nodes failed, Not all workers node are labeled !")
+
+    def _clone_operator(self):
+        """
+        clone the benchmark-operator repo into temp directory
+
+        """
+        self.dir = tempfile.mkdtemp(prefix=f"{BMO_NAME}_")
+        try:
+            log.info(f"Cloning {BMO_NAME} in {self.dir}")
+            git_clone_cmd = f"git clone -b {self.branch} {self.repo} --depth 1"
+            run(git_clone_cmd, shell=True, cwd=self.dir, check=True)
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error(f"Error during cloning of {BMO_NAME} repository")
+            raise cf
+
+    def deploy(self):
+        """
+        Deploy the benchmark-operator
+
+        """
+        log.info("Deploy the benchmark-operator project")
+        try:
+            run("make deploy", shell=True, check=True, cwd=self.dir)
+        except Exception as ex:
+            log.error(f"Failed to deploy benchmark operator : {ex}")
+
+        log.info("Wait for the benchmark-operator deployment be available")
+        try:
+            cmd = f'wait --for=condition=available "{BMO_DEPLOYMENT}" -n {BMO_NAME} --timeout=300s'
+            self.pod_obj.exec_oc_cmd(cmd)
+
+            # At this point the benchmark operator pod is ready, but we need to
+            # verifying that all containers in the pod are ready
+            while True:
+                OK = 1
+                result = self.pod_obj.exec_oc_cmd(f"get pod -n {BMO_NAME} -o json")
+                for cnt in (
+                    result.get("items")[0].get("status").get("containerStatuses")
+                ):
+                    if not cnt.get("ready"):
+                        OK = 0
+                if not OK:
+                    log.warning("Benchmark Operator is not ready yet")
+                    time.sleep(3)
+                else:
+                    break
```

Comment:
@Avilir this loop might cause automation wait for long time if pod is not at all in ready state. I would suggest to add another break for failure.

Reply:
Ok, will fix it

Reply:
@ramkiperiy , since this loop is in the try-except section, it will come tom this loop only if it is in Ready state, if not Ready the except will prevent from this loop to run.
that said, i will add another exit point from the loop